stdlib/bytes.ml

```diff
@@ -97,19 +97,30 @@ let iter f a =
 let iteri f a =
   for i = 0 to length a - 1 do f i (unsafe_get a i) done
 
+let () = assert (3 * Sys.max_string_length <= max_int)
```

Comment:
@dbuenzli: it seems helpful for the test to fail in BuckleScript, since the `concat` code relies on an assumption that doesn't hold there.